### ISSUE TITLE
F# requiring assembly binding redirects for FSharp.Core

### ISSUE THREAD

**Issue by kolektiv** [PerfDotNet/BenchmarkDotNet#67]
I'm trying to benchmark some F# code. However, to work properly, the F# application requires an assembly binding redirect (App.config):

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="FSharp.Core" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-999.999.999.999" newVersion="4.4.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
  <startup>
    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2" />
  </startup>
</configuration>
```

The `supportedRuntime` parts are not desirable of course, but if the assembly binding isn't done, the auto generated application will fail at runtime. The app.config generated won't include anything like this.

I'm not sure how to work around it, or suggest it could be fixed. Taking the app.config from the host app if one exists seems vaguely possible, but a bit random and risky, there's no reason why it should really be correct. Being able to specify app.config elements might be possible I guess...

**Comment by kolektiv** [comment]
If it helps to see this in-situ, the project i'm trying to benchmark is here: https://github.com/kolektiv/chiron/tree/benchmarks - https://github.com/kolektiv/chiron/tree/benchmarks/tests/Chiron.Benchmarks being the relevant project.

If you want to try and run that, you'll need to restore the packages first. The build.cmd in the root should sort all of that out and get a build done to start with.

**Comment by kolektiv** [comment]
Last bit of info, to confirm the thinking - if I build a version of BenchmarkDotNet which includes that assembly binding in the templated app.configs, the benchmark will run fine. I'm not sure how easy it will be to come up with an elegant fix though, I'll think on it.

**Comment by adamsitnik** [comment]
Assembly binding redirects can be auto-generated. I used `<AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
` to get it working.
[Some explanation](http://mohitshiwach.blogspot.com/2014/11/assembly-binding-redirection-with-net.html)

**Comment by kolektiv** [comment]
This still doesn't seem to actually work for me. Looking at the app.config generated there are no binding redirects, and without them the program simply fails. Auto generate binding redirects doesn't appear to do anything in this case!

**Comment by kolektiv** [comment]
The actual example where I'm seeing this: https://github.com/kolektiv/freya-benchmarks

Running build.cmd should pull down all appropriate dependencies, etc.

**Comment by adamsitnik** [comment]
It is quite long story. For 0.9.5 I [added fix for that](https://github.com/PerfDotNet/BenchmarkDotNet/commit/2645ef3c25225730d5f255d2e4dde8a6c10da78d). The problem was that you needed MSBuild 12/14 to get it working, which our .NET 4.5/4.6 taget was referencing. If you added depndency to our .NET 4.0 lib, then it did not work because we used MSbuild.v4 there.

Then we moved from dnx to dotnet cli for 0.9.7 and we were not able to target different versions of the same assembly, so we have removed our 4.5 and 4.6 targets. So it stopped working again for 0.9.7.

Now we will do this on our own, independent from installed MSBuild version. However if you don't want to wait for 0.9.8 which will have the fix you can downgrade to 0.9.5/0.9.6 for now.
